src/GraphQL/Server.hs

```diff
@@ -83,20 +82,34 @@ class (MonadThrow m, MonadIO m) => HasGraph m a where
   type Handler m a
   buildResolver :: Handler m a -> CanonicalQuery -> m GValue.Value
 
--- | The ReadValue instance converts AST.Value types like ValueInt to
--- the type expected by the handler function. It's the boundary
--- between incoming types and your custom application Haskell types.
-class ReadValue a where
-  -- | Convert the already-parsed value into the type needed for the
-  -- function call.
-  readValue :: AST.Value -> Either Text a
 
-  -- | valueMissing is a separate function so we can provide default
-  -- values for certain cases. E.g. there is an instance for @@Maybe a@@
-  -- that returns Nothing if the value is missing.
-  valueMissing :: AST.Name -> Either Text a
-  valueMissing name' = Left ("Value missing: " <> AST.getNameText name')
+-- | Specify a default value for a type in a GraphQL schema.
+--
+-- GraphQL schema can have default values in certain places. For example,
+-- arguments to fields can have default values. Because we cannot lift
+-- arbitrary values to the type level, we need some way of getting at those
+-- values. This typeclass provides the means.
+--
+-- To specify a default, implement this typeclass.
+--
+-- The default implementation is to say that there *is* no default for this
+-- type.
+class Defaultable a where
+  -- | defaultFor returns the value to be used when no value has been given.
+  defaultFor :: AST.Name -> Maybe a
+  defaultFor _ = empty
+
+-- | Called when we're missing a value of type @a@ with the given 'AST.Name'.
```

Comment:
Maybe something like "Called when the schema expects an input argument @name@ but @name@ has not been provided in the query" ?